Support for procedure method override

jlalmes opened the pull request:
Closes https://github.com/trpc/trpc/issues/1939.

## Support for procedure method override

By passing an optional `method?: 'GET' | 'POST'` property into `TRPCRequestOptions`, you can override the HTTP transport method used by queries/mutations. This is great for circumventing URL length limits when sending large input payloads to `query` procedures (there are use-cases for sending a `mutation` via GET too).

### Implementation
If `method` override is present the original `type` is added as a query parameter (i.e. `?type=query`). The `@trpc/server` will then resolve the `input` as the `POST` body but proceed to handle the request as a `query`. 

### Example
Query as POST: `client.query('getData', { bigPayload: [...] }, { method: 'POST' })`

### Additional
`httpLink` & `httpBatchLink` will now throw an error when sent a `subscription` operation.

mmkal commented on packages/client/src/internals/httpRequest.ts:
```diff
@@ -15,17 +14,17 @@ function arrayToDict(array: unknown[]) {
     runtime: LinkRuntimeOptions;
-    type: ProcedureType;
+    type: 'query' | 'mutation';
+    method: 'query' | 'mutation';
```
I think `'get' | 'post'` Are more intuitive here don’t you think?

jlalmes replied:
Option A
```suggestion
    method: 'GET' | 'POST';
```
Option B
```suggestion
    typeOverride: 'query' | 'mutation';
```

Thoughts?

mmkal replied:
I prefer A, it more directly tells the end user what the property will actually do. Aside from changing get to post it doesn’t somehow turn a query into a mutation. It’s purely an http thing right?

jlalmes commented:
Just updated description and added tests - ready for review!

KATT reviewed (commented):
This could be used for DoS-attacks, could we make it opt-out on the server the same wait we can opt-out of batching?

sachinraja reviewed (approved):
Looks good - I agree with Alex's comment and just left a few code quality comments. I think we should couple this with good docs to prevent unnecessary usage and hopefully the new transformer API can also help here. I still think it's useful for production apps despite opening up DoS attacks and some providers allow you to cache POST requests.

jlalmes commented:
* Normalized method in `TRPCClient.ts`.
* Deduped HTTP subscription error message.
* Added opt-out for server via `methodOverride.enabled: false`.
* Fixed tests & added test for opt-out.

KATT commented on packages/client/src/internals/executeChain.ts:
```diff
@@ -62,6 +63,8 @@ export function executeChain<
   }
+
   walk({ index: 0, op: opts.op, stack: [] });
+
```
```suggestion
```

KATT commented:
Closing because of the many conflicts it has, feel free to pick it up though
